models/branches.go

```diff
@@ -209,6 +210,131 @@ func (protectBranch *ProtectedBranch) GetProtectedFilePatterns() []glob.Glob {
 	return extarr
 }
 
+// MergeBlockedByProtectedFiles returns true if merge is blocked by protected files change
+func (protectBranch *ProtectedBranch) MergeBlockedByProtectedFiles(pr *PullRequest) bool {
```

Comment:
I'm concerned that this is expensive to run and should be cached by the pr checker

Reply:
Yes, it's meaningful, but if using this way, the result can't be changed on time when you change the protected files setting. I wonder...

Reply:
@zeripath  How about this way? Thanks.
ea6716541a460df58fcfd659a5a31492f591b448